packages/react-next/src/components/Autofill/Autofill.tsx

```diff
@@ -308,42 +298,64 @@ export class Autofill extends React.Component<IAutofillProps, IAutofillState> im
   private _updateValue = (newValue: string, composing: boolean) => {
     // Only proceed if the value is nonempty and is different from the old value
     // This is to work around the fact that, in IE 11, inputs with a placeholder fire an onInput event on focus
-    if (!newValue && newValue === this._value) {
+    if (!newValue && newValue === this.value) {
       return;
     }
-    this._value = this.props.onInputChange ? this.props.onInputChange(newValue, composing) : newValue;
+
     this.setState(
       {
-        displayValue: this._getDisplayValue(this._value, this.props.suggestedDisplayValue),
+        inputValue: this._getInputChange(newValue, composing),
       },
-      () => this._notifyInputChange(this._value, composing),
+      () => this._notifyInputChange(newValue, composing),
     );
   };
 
-  /**
-   * Returns a string that should be used as the display value.
-   * It evaluates this based on whether or not the suggested value starts with the input value
-   * and whether or not autofill is enabled.
-   * @param inputValue - the value that the input currently has.
-   * @param suggestedDisplayValue - the possible full value
-   */
-  private _getDisplayValue(inputValue: string, suggestedDisplayValue?: string): string {
-    let displayValue = inputValue;
-    if (
-      suggestedDisplayValue &&
-      inputValue &&
-      this._doesTextStartWith(suggestedDisplayValue, displayValue) &&
-      this._autoFillEnabled
-    ) {
-      displayValue = suggestedDisplayValue;
+  private _getDisplayValue(): string {
+    if (this._autoFillEnabled) {
+      return _getDisplayValue(this.value, this.props.suggestedDisplayValue);
     }
-    return displayValue;
+
+    return this.value;
   }
 
-  private _doesTextStartWith(text: string, startWith: string): boolean {
-    if (!text || !startWith) {
-      return false;
+  private _getInputChange = (newValue: string, composing: boolean) => {
+    const { onInputChange } = this.props;
+    if (onInputChange) {
+      return onInputChange?.(newValue, composing) || '';
+    }
+    return newValue;
+  };
+
+  private _getControlledValue(): string | undefined {
+    const { value } = this.props;
+    if (value === undefined || typeof value === 'string') {
```

Comment:
Can you also update the type of `value` in the types file? That's also good for documentation (people get confused when essential props are implicitly inherited).

Reply:
I'm still not sure in what cases it would even be possible/reasonable for the value to be another type, given the expected usage of this component.

Reply:
Yeah I'll add it.